components/sup/src/main.rs

```diff
@@ -233,44 +233,41 @@ fn sub_run(m: &ArgMatches, launcher: LauncherCli) -> Result<()> {
     set_supervisor_logging_options(m);
 
     let cfg = mgrcfg_from_matches(m)?;
-    if Manager::is_running(&cfg)? {
-        process::exit(OK_NO_RETRY_EXCODE);
-    } else {
```

Comment:
To clarify, this is what was ultimately causing the lack of restarting, right? The launcher would try to restart the supervisor, the new Supervisor process would see that there's already a Launcher running, and then shut itself down?

Reply:
Yup

Reply:
Nice sleuthing on that one 👏